Microsoft/MixedRealityToolkit-Unity
# Add binary nuget package compilation

Overview
---

This adds csproj files that compile to binary files and get packed into nupkgs. 

Current progress is not complete, but should be reduced to minor issues. The goal of this change: 

* Understand folder structure of new solution. 
* Commit to technology investment 
* Move forward on packaging

Instructions to build and package: run `dotnet pack` in /Build folder. Tested with version `2.1.402` of dotnet core.

Version number still needs to be resolved, so the current one is a placeholder. #2888  has a discussion on this topic. 


Testing done: 

* Use NuGetForUnity to consume local nupkg
* Make sure the "Configure..." menu item exists and can be clicked

Currently default profiles still don't seem to resolve correctly. Since this is a feature branch, I consider that acceptable to address in another PR as I continue to test and move forward.  I have not tested that the solution deploys and behaves correctly, only compilation problems have been resolved.
